src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -163,26 +158,28 @@ private Map<String, List<String>> getERICTokenPermissions(HttpServletRequest req
         return permissions;
     }
 
-    private Optional<String> getCompanyNumberInTransaction(HttpServletRequest request, HttpServletResponse response, String transactionId) {
+    private Optional<String> getCompanyNumberInTransaction(HttpServletRequest request, String transactionId) throws ServiceException {
         String passthroughHeader = request.getHeader(ApiSdkManager.getEricPassthroughTokenHeader());
         String reqId = request.getHeader(ERIC_REQUEST_ID_KEY);
         var logMap = new HashMap<String, Object>();
         logMap.put(TRANSACTION_ID_KEY, transactionId);
-        try {
-            final var transaction = transactionService.getTransaction(transactionId, passthroughHeader, reqId);
-            logMap.put(COMPANY_NUMBER_KEY, transaction.getCompanyNumber());
-            ApiLogger.debugContext(reqId, "Transaction successfully retrieved " + transactionId, logMap);
-            if (StringUtils.isEmpty(transaction.getCompanyNumber())) {
-                response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-                return Optional.empty();
-            }
-            return Optional.of(transaction.getCompanyNumber());
-        } catch (ServiceException e) {
-            ApiLogger.errorContext(reqId, "Error retrieving transaction " + transactionId, e, logMap);
-            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+
+        final var transaction = transactionService.getTransaction(transactionId, passthroughHeader, reqId);
+        if (transaction == null) {
             return Optional.empty();
         }
+        logMap.put("company_number_in_transaction", getNotNull(transaction.getCompanyNumber()));
+        ApiLogger.debugContext(reqId, "Transaction successfully retrieved " + transactionId, logMap);
+        return Optional.ofNullable(transaction.getCompanyNumber());
 
     }
 
+    private String getNotNull(String aString) {
```

Comment:
As mentioned above, if logic re-organised slightly, I don't think this method is required.

Reply:
I've kept this method and using it in 2 places to check for null in the LogMap entry.

Reply:
I'm just not sure that doing this conversion is correct - with it, if a `null` value or an empty string is in the log map, they both appear as the same thing in the output, e.g.

`"company_number_in_transaction":""`

It could be useful to distinguish between the two. I think just not doing the conversion would be fine (in this case, the null value set in the log map is suppressed from the output), but if continuing to keep this method, I would name it something like `convertNullValuesForLogMapOutput` and return "NULL" rather than the empty string.